swap/src/protocol/alice/event_loop.rs

```diff
@@ -301,30 +300,32 @@ where
         &mut self,
         btc: bitcoin::Amount,
         monero_wallet: Arc<monero::Wallet>,
-    ) -> Result<monero::Amount> {
+    ) -> Result<Result<monero::Amount, spot_price::Error>> {
+        if self.resume_only {
+            return Ok(Err(spot_price::Error::MaintenanceMode));
+        }
```

Comment:
I find this approach a bit weird, i.e. return an error to indicate that we don't care about the price requests. 
Why did you move the `self.resume_only` out of the loop?

Reply:
To keep all logic related to `spot-price` requests in a dedicated function and not scatter them over different places (i.e. event loop vs. this function. The function is called `handle_spot_price_request` and this check is part of that I'd say. 
This should help us move this logic into a dedicated NetworkBehaviour at some point.

Reply:
Note: This comment became somewhat outdated with https://github.com/comit-network/xmr-btc-swap/pull/461/commits/24da186be998518eeb94db4ef61eb3f8c6f3b904